### PR TITLE
feat: Add draft json schema validation and serde model

### PR DESCRIPTION
**Pull request by seakayone**
- add json schema draft and current metadata as json
- add test which fails if json data is not matching the draft schema
-  fix test and remove code duplication 
- add just recipes for formatting and watching changes and run tests
- backport https://github.com/dasch-swiss/dsp-meta-svc/pull/145

### CODE REVIEW

File: docs/domain_model/schema-metadata.json [hunk 1]
```diff
@@ -304,6 +307,28 @@
...
+          "type": "string",
+          "description": "The citation form of the publication."
+        },
+        "url": {
+          "description": "The URLs of the publication, preferably PIDs like DOI or ARK.",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "$ref": "#/definitions/url"
+          }
```
Comment (mpro7) [inline docs/domain_model/schema-metadata.json:324]: What kind of items are this?

File: dsp-meta/resources/schema-metadata-draft.json [hunk 2]
```diff
@@ -0,0 +1,850 @@
+{
```
Comment (mpro7) [inline dsp-meta/resources/schema-metadata-draft.json:1]: Is this draft something different than existing already `schema-metadata.json`?

**Review by mpro7 — APPROVED** [review]

File: dsp-meta/resources/schema-metadata-draft.json (hunk 2 above)
Comment (seakayone) [reply dsp-meta/resources/schema-metadata-draft.json:1]: Yes, in our last meeting we decided to  mark all properties as optional which are not present for all json.

File: docs/domain_model/schema-metadata.json (hunk 1 above)
Comment (seakayone) [reply docs/domain_model/schema-metadata.json:324]: Its part of the JSON schema definition: https://json-schema.org/specification

Comment (seakayone) [reply docs/domain_model/schema-metadata.json:324]: These changes are the backport of https://github.com/dasch-swiss/dsp-meta-svc/pull/145 in the "old" service

**Pull request merged**